babel/core.py

```diff
@@ -1190,22 +1233,23 @@ def parse_locale(identifier: str, sep: str = '_') -> tuple[str, str | None, str
     if parts:
         raise ValueError(f"{identifier!r} is not a valid locale identifier")
 
-    return lang, territory, script, variant
+    return lang, territory, script, variant, modifier
```

Comment:
If you use `partition`:
```suggestion
    return lang, territory, script, variant, (modifier or None)
```

Reply:
Yeah, related to the change above. Makes tests pass.